client/src/app/site/motions/services/workflow-repository.service.ts

```diff
@@ -70,4 +71,23 @@ export class WorkflowRepositoryService extends BaseRepository<ViewWorkflow, Work
         });
         return states;
     }
+
+    /**
+     * Returns all workflowStates that cover the list of viewMotions given
+     * @param motions
+     */
+    public getWorkflowStatesForMotions(motions: ViewMotion[]): WorkflowState[] {
+        const workFlowList: number[] = [];
+        let states: WorkflowState[] = [];
+        motions.forEach(vm => {
```

Comment:
Your goal is to get a unique list of workflow ids, right?
```
const workflowIds = motions.map(motion => motion.workflow_id).filter((value, index, self) => self.indexOf(value) === index);
```
(see https://stackoverflow.com/questions/1960473/get-all-unique-values-in-a-javascript-array-remove-duplicates, very nice ;) )